### ISSUE TITLE
Bug: Namespace import fails for attributes on local functions

### ISSUE THREAD

**Issue by bkoelman** [dotnet/roslyn#16245]
**Version Used**: 
Microsoft.CodeAnalysis v2.0.0-rc2

**Steps to Reproduce**:
Run the next program in VS 2017 RC2 as a .NET Framework console application that targets .NET 4.6.1:
```csharp
using System;
using System.Linq;
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.CSharp;
using Microsoft.CodeAnalysis.CSharp.Syntax;
using Microsoft.CodeAnalysis.Editing;
using Microsoft.CodeAnalysis.Formatting;
using Microsoft.CodeAnalysis.Simplification;

namespace FormatterSimplifierTest
{
    class Program
    {
        static void Main()
        {
            const string attributeSource = @"
                namespace Z
                {
                    public class SomeAttribute : System.Attribute
                    {
                    }
                }";

            const string source = @"
                namespace X
                {
                    class C
                    {
                        static void M(string p1)
                        {
                            void N(string p2)
                            {
                            }

                            N(string.Empty);
                        }
                    }
                }";

            Document document = new AdhocWorkspace()
                .AddProject("TestProject", LanguageNames.CSharp)
                .AddMetadataReferences(new[] {MetadataReference.CreateFromFile(typeof(object).Assembly.Location)})
                .WithCompilationOptions(new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary))
                .AddDocument("Source.cs", source)
                .Project
                .AddDocument("Attribute.cs", attributeSource)
                .Project
                .Documents.First();

            SemanticModel model = document.GetSemanticModelAsync().Result;

            if (model.Compilation.GetDiagnostics().Any())
            {
                throw new Exception(string.Join(Environment.NewLine, model.Compilation.GetDiagnostics()));
            }

            IMethodSymbol mSymbol = (IMethodSymbol) model.Compilation.GetTypeByMetadataName("X.C").GetMembers("M").Single();
            MethodDeclarationSyntax mSyntax = (MethodDeclarationSyntax) mSymbol.DeclaringSyntaxReferences[0].GetSyntax();
            ParameterSyntax p1Syntax = mSyntax.DescendantNodes().OfType<ParameterSyntax>().First();

            LocalFunctionStatementSyntax nSyntax = mSyntax.DescendantNodes().OfType<LocalFunctionStatementSyntax>().Single();
            ParameterSyntax p2Syntax = nSyntax.DescendantNodes().OfType<ParameterSyntax>().Single();

            INamedTypeSymbol someAttributeSymbol = model.Compilation.GetTypeByMetadataName("Z.SomeAttribute");

            Console.WriteLine(AddAttributeTo(p1Syntax, someAttributeSymbol, document));
            Console.WriteLine("---");
            Console.WriteLine(AddAttributeTo(p2Syntax, someAttributeSymbol, document));
        }

        private static string AddAttributeTo(SyntaxNode target, INamedTypeSymbol attributeSymbol, Document document)
        {
            SyntaxAnnotation namespaceImportAnnotation = new SyntaxAnnotation();

            DocumentEditor editor = DocumentEditor.CreateAsync(document).Result;

            // Add Z.SomeAttribute
            SyntaxNode attributeSyntax =
                editor.Generator.Attribute(editor.Generator.TypeExpression(attributeSymbol))
                    .WithAdditionalAnnotations(Simplifier.Annotation, Formatter.Annotation, namespaceImportAnnotation);
            editor.AddAttribute(target, attributeSyntax);
            Document documentWithAttribute = editor.GetChangedDocument();
            string documentWithAttributeText = documentWithAttribute.GetSyntaxTreeAsync().Result.ToString();

            // Add namespace import.
            Document documentWithImport =
                ImportAdder.AddImportsAsync(documentWithAttribute, namespaceImportAnnotation).Result;
            string documentWithImportText = documentWithAttribute.GetSyntaxTreeAsync().Result.ToString();

            Document documentSimplified = Simplifier.ReduceAsync(documentWithImport).Result;
            string documentSimplifiedText = documentSimplified.GetSyntaxTreeAsync().Result.ToString();

            Document documentFormatted = Formatter.FormatAsync(documentSimplified).Result;
            string documentFormattedText = documentFormatted.GetSyntaxTreeAsync().Result.ToString();

            return documentFormattedText;
        }
    }
}
```

**Expected Behavior**:
A namespace import is added for `Z.SomeAttribute` on both method and local function parameter.
```

using Z;

namespace X
{
    class C
    {
        static void M([Some] string p1)
        {
            void N(string p2)
            {
            }

            N(string.Empty);
        }
    }
}
---

using Z;

namespace X
{
    class C
    {
        static void M(string p1)
        {
            void N([Some] string p2)
            {
            }

            N(string.Empty);
        }
    }
}
```

**Actual Behavior**:
Namespace import fails silently for `Z.SomeAttribute` on local function `X.C.M.N`. It seems that because of this, the simplifier leaves the attribute untouched.
```

using Z;

namespace X
{
    class C
    {
        static void M([Some] string p1)
        {
            void N(string p2)
            {
            }

            N(string.Empty);
        }
    }
}
---

namespace X
{
    class C
    {
        static void M(string p1)
        {
            void N([global::Z.SomeAttribute] string p2)
            {
            }

            N(string.Empty);
        }
    }
}
```

**Comment by gafter** [comment]
I suspect that attributes are not allowed on parameters to local functions.

**Comment by agocke** [comment]
@bkoelman @gafter Correct. This should properly error after when PR https://github.com/dotnet/roslyn/pull/16498 is merged.

**Comment by agocke** [comment]
Appears to be fixed by #16498
